File: hw2/RaceMonitor_Java/src/Messages/LocationUpdate.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Messages;

import Exceptions.ApplicationException;
import Racedata.AthleteRaceStatus;
/**
 *
 * @author Shimin
 * @author Stephen Clyde
 */
public class LocationUpdate extends AthleteUpdate {
    private double locationOnCourse;
    
    public LocationUpdate() {
        super(AthleteRaceStatus.OnCourse);
    }
    
    public LocationUpdate(String[] properties) throws ApplicationException {
        super (AthleteRaceStatus.OnCourse, properties);
        
        // the cs file LocationUpdate.cs on line 14 might have a bug here
        if (properties.length != 4 || !properties[0].equals(AthleteRaceStatus.OnCourse.toString()))
            throw new ApplicationException("Invalid properties");
        
        locationOnCourse = Double.parseDouble(properties[3]);
    }
    
    @Override
    public String toString() {
        return super.toString() + "," + locationOnCourse;
    }
    
    public void setLocationOnCourse (double value) { locationOnCourse = value; }

    public double getLocationOnCourse () {
        return locationOnCourse;
    }
}

File: hw2/RaceMonitor_CSharp/RaceData/Messages/LocationUpdate.cs
using System;

namespace RaceData.Messages
{

    public class LocationUpdate : AthleteUpdate
    {
        public double LocationOnCourse { get; set; }

        public LocationUpdate() : base(AthleteRaceStatus.OnCourse) { }

        public LocationUpdate(string[] properties) : base(AthleteRaceStatus.OnCourse, properties)
        {
            if (properties.Length != 4 || properties[0] != AthleteRaceStatus.OnCourse.ToString())
                throw new ApplicationException("Invalid properties");

            LocationOnCourse = Convert.ToDouble(properties[3]);
        }

        public override string ToString()
        {
            return $"{base.ToString()},{LocationOnCourse}";
        }
    }
}
